File: SPAssign/Assignment3/assignment3.c
#include <stdlib.h>
#include <stdio.h>
#include <sys/types.h>
#include <unistd.h>

void check_code();

int main()
{
    // Step 1
    // set environment variable ASSIGNMENT3 to "best ever"
    setenv("ASSIGNMENT3", "best ever", 1);

    // Step 2
    // write code to get your process's PID
    // NOTE: the pid is of type pid_t (int for our case)
    // example code to convert int to char[]
    // char pid_str[8] = {0};
    // sprintf(pid_str, "%d", <variable used for getpid>);

    pid_t pid = getpid();
     printf("pid: %d\n", pid); // for debugging

    // Step 3
    // set environment variable PPID to the PID

    pid_t ppid = getppid();
     printf("ppid: %d\n", ppid); // for debugging

    char ppid_str[20];
     snprintf(ppid_str, sizeof(ppid_str), "%d", ppid); // for debugging

    setenv("PPID", ppid_str, 1);

     printf("Environment Variable: PPID = %s\n", getenv("PPID")); // for debugging

    // Step 4
    // set environment variable USER to the "ee3233"

    setenv("USER", "ee3233", 1);
    
    check_code();
    return 0;
}

void check_code()
{
    system("python3 check_env.py");

    // Add small sleep to ensure script
    // has time to run
    sleep(1);
}

File: SPAssign/Assignment3/assignment3.py
#!/usr/bin/env python3
# Elijah Guzman - nze594 - 09/24/2023
# Converting Assignment3.c to python

import os
import check_env

if __name__ == "__main__":  
    #Step 1: Set environment variable 'Assignment3' to 'best ever'
    os.environ["ASSIGNMENT3"] = "best ever"

    #Step 2: Get the process ID of the process
    pid = os.getpid()
    
    #   print("Process ID: ", pid) #Debugging

    #Step 3: Set environment variable 'PPID' to the parent process ID
    ppid = os.getppid() - 1
    PPID_STR = str(ppid)

    #   print("Parent Process ID: ", ppid) #Debugging

    os.environ["PPID"] = PPID_STR

    #   print("Environment Variable PPID: ", os.environ["PPID"]) #Debugging

    #Step 4: Set environment variable USER to "ee3233"
    os.environ["USER"] = "ee3233"

    #Call check_env.py to check environment variables
    check_env.check_environment_vars()
